devito/ops/node_factory.py

```diff
@@ -38,7 +39,8 @@ def new_ops_arg(self, indexed):
 
         if ops_arg_id not in self.ops_args:
             # Create the indexed object
-            ops_arg = Array(name=ops_arg_id,
+            ops_arg = Array(is_Write,
```

Comment:
If this were a subclass of Scalar, instead of Array (see also my other comment below!), then you could drive the generation of const by simply providing a proper value for the [is_const flag](https://github.com/opesci/devito/blob/master/devito/types/basic.py#L383), which eventually gets used to generate or not the `const` modifier, see [here](https://github.com/opesci/devito/blob/master/devito/types/basic.py#L301)